File: callperl.scala

import org.perl.inline.java._

class HelpMePerl {
    val pi = new InlineJavaPerlCaller

    def xmatch(target: String, pattern: String): Boolean = {
        val b = pi.eval("'" + target + "' =~ /" + pattern + "/", classOf[Boolean])
        return b.asInstanceOf[Boolean]
    }
}

File: callperl.pl
#!/usr/bin/env perl

use strict;
use warnings;
use Inline(
    'Java' => 'STUDY',
    STUDY     => ['HelpMePerl'],
    AUTOSTUDY => 1,
    CLASSPATH => $ENV{SCALA_HOME} . '/lib/scala-library.jar:.'
);

my $h = HelpMePerl->new();

my $z = $h->xmatch('abcdefg','xxd');

print "$z\n";
